dask/array/core.py

```diff
@@ -835,6 +836,24 @@ def __getitem__(self, index):
 
         return Array(merge(self.dask, dsk), out, chunks, dtype=self._dtype)
 
+    def _vindex(self, key):
+        if (not isinstance(key, tuple) or
+           not len([k for k in key if isinstance(k, (np.ndarray, list))]) >= 2 or
+           not all(isinstance(k, (np.ndarray, list)) or k == slice(None, None)
+                   for k in key)):
+            raise IndexError("vindex expects only lists and full slices\n"
```

Comment:
to be clear: this will fail for multi-dimensional indexers, right? if so, we should catch that.